File: src/2025/day09/src/test/java/ca/terrylockett/aoc2025/day09/TestDay09.java
package ca.terrylockett.aoc2025.day09;

import ca.terrylockett.aoccommon.input.PuzzleInput;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TestDay09 {

	static final String TEST_FILE_NAME = "test-input.txt";

	static String testInput = "";

	@BeforeAll
	static void setup() {
		testInput = PuzzleInput.getInput(TEST_FILE_NAME).orElseThrow();
	}

	@Test
	void testDay09part1() {
		assertEquals(50L, Day09Runner.part1(testInput));
	}
}

File: src/2023/day09/src/test/kotlin/ca/terrylockett/aoc2023/day09/TestDay09.kt
package ca.terrylockett.aoc2023.day09

import ca.terrylockett.aoccommon.resources.Resources
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TestDay09 {

	@Test
	fun part1() {
		val inputFilePath = Resources.getInputFilePath("test-input.txt").orElseThrow()
		assertEquals(114, part1(inputFilePath))
	}

	@Test
	fun part2() {
		val inputFilePath = Resources.getInputFilePath("test-input.txt").orElseThrow()
		assertEquals(2, part2(inputFilePath))
	}
}
